osquery/utils/windows/lzxpress.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/logger/logger.h>
+#include <osquery/utils/expected/expected.h>
+#include <osquery/utils/windows/lzxpress.h>
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/algorithm/string.hpp>
+
+namespace osquery {
+ExpectedDecompressData decompressLZxpress(std::vector<char> prefetch_data,
+                                          unsigned long size) {
+  static HMODULE hDLL =
+      LoadLibraryExW(L"ntdll.dll", NULL, LOAD_LIBRARY_SEARCH_SYSTEM32);
```

Comment:
This call is not freed anywhere. Recommend instead to use `GetModuleHandle` instead. ntdll.dll will already be loaded as a result of the import address table. You can assume the loaded ntdll.dll is in system32 (otherwise, the process has already loaded an incorrect ntdll.dll, and whether your feature uses the correct one or not won't matter).

Reply:
Thats good to know, thanks
Switched to `GetModuleHandle`